src/engine/Server.cpp

```diff
@@ -45,12 +45,25 @@ void Server::initialize(const string& indexBaseName, bool useText,
 // _____________________________________________________________________________
 void Server::run(const string& indexBaseName, bool useText, bool usePatterns,
                  bool usePatternTrick, bool loadAllPermutations) {
-  // First set up the HTTP server, so that it binds to the socket, and
-  // the "socket already in use" error appears quickly.
+  using namespace ad_utility::httpUtils;
+
+  // Function that handles a request asynchronously. Catches any exception
+  // thrown inside of the `process` function and sends a "HTTP/1.1 400 Bad
+  // Request" response to the client with the error message from the exception.
   auto httpSessionHandler =
       [this](auto request, auto&& send) -> boost::asio::awaitable<void> {
-    co_await process(std::move(request), send);
+    try {
```

Comment:
``` 
std::optional<boost::asio::awaitable<void>> errorSender;
try { .... } catch (const Exception& e) {
  errorSender = send(createBadRequest(....
}
if (errorSender) {
  co_return co_await errorSender.value(); // possibly with std::move
}
```
If there's a problem with that, store the return value of `createBadRequest` in an optional and then `co_await send(...)` after the catch.

co_await inside catch is forbidden by the c++-standard.

Reply:
And I am not sure if we also need "sendWithCors" here.
If so, then set it up here, and pass it as the second argument to the `process` function.